From 0c571dd07708fde7e668d1890b968c292bcda699 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Sat, 22 Jun 2019 09:58:29 +0200
Subject: [PATCH] link with -latomic on mips(el), armel, powerpc

needed since 0d04315c17a6df9f971237d45d9e5e8af765dd17 since it fails
without, see
https://buildd.debian.org/status/logs.php?pkg=libreoffice&ver=1%3A6.3.0~beta2-1&suite=experimental

Change-Id: Ie91b19c0bd534b4391741388fc93494ff5a1ddf0
---
 solenv/gbuild/platform/LINUX_ARM_GCC.mk     | 2 ++
 solenv/gbuild/platform/LINUX_GODSON_GCC.mk  | 2 ++
 solenv/gbuild/platform/LINUX_POWERPC_GCC.mk | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/solenv/gbuild/platform/LINUX_ARM_GCC.mk b/solenv/gbuild/platform/LINUX_ARM_GCC.mk
index cccf5772b3b4..e52cf2bb8950 100644
--- a/solenv/gbuild/platform/LINUX_ARM_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_ARM_GCC.mk
@@ -13,4 +13,6 @@ gb_COMPILEROPTFLAGS := -Os
 
 include $(GBUILDDIR)/platform/linux.mk
 
+gb_LinkTarget_LDFLAGS += -latomic
+
 # vim: set noet sw=4:
diff --git a/solenv/gbuild/platform/LINUX_GODSON_GCC.mk b/solenv/gbuild/platform/LINUX_GODSON_GCC.mk
index 976aaedf97fe..f16f98530e2b 100644
--- a/solenv/gbuild/platform/LINUX_GODSON_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_GODSON_GCC.mk
@@ -13,4 +13,6 @@ gb_COMPILEROPTFLAGS := -Os
 
 include $(GBUILDDIR)/platform/linux.mk
 
+gb_LinkTarget_LDFLAGS += -latomic
+
 # vim: set noet sw=4:
diff --git a/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk b/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk
index 1880b3bad1e4..48fa395cfada 100644
--- a/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk
+++ b/solenv/gbuild/platform/LINUX_POWERPC_GCC.mk
@@ -12,4 +12,6 @@ gb_CPUDEFS += -DPPC
 
 include $(GBUILDDIR)/platform/linux.mk
 
+gb_LinkTarget_LDFLAGS += -latomic
+
 # vim: set noet sw=4:
-- 
2.20.1

